### ISSUE TITLE
Expanding support to more file types

### ISSUE THREAD

**Issue by vguillet** [brianpetro/obsidian-smart-connections#148]
Hey, absolutely in love with the plugin so far. One of the main limitations I would say so far is the lack of PDF support, images text extraction, etc... A number of plugins might help with this, such as the text-extractor plugin (leveraged by the omnisearch plugin for this very purpose), but having the ability to also get responses based on images and pdfs would go such a long way to improving the chat quality (in my case at the very least XD). Cant wait to see where this plugin heads next non-the-less!

**Comment by brianpetro** [comment]
Hey @vguillet 

Great suggestion. I think PDF support would be a great addition.

Already peaked at https://github.com/scambier/obsidian-text-extractor but I'll have to do some more investigating to better understand how an integration might work.

For images, I think we just need to wait for the release of GPT-4's multimodal capabilities. That should allow an easy integration.

Thanks again for your continued helpful feedback!

Brian 🌴

**Comment by engr-lynx** [comment]
Agree, especially pdf support. A lot of references are in pdf. Having the ability to make smart connections between these and your own notes is a game-changer.

**Comment by dannygithub1** [comment]
Agree - this would be a complete changer. Often my PDFs are themselves pieces of work I produce which do not get retyped in Obsidian.

**Comment by brianpetro** [comment]
In the meantime, check out [this video](https://www.youtube.com/watch?v=909wdjg71fM&t=4s) for how the Smart Connect GPT can help create notes that semantically represent the PDF such that it's discoverable in Smart Connections 🌴
